### PR TITLE
Created a function for Relative Observation Trends

### PR DESCRIPTION
**Pull request by antpn**
Check out the code. It should probalby be inserted into some other R-file. 
It seems to work! :) 
RelObsTrends Bombus sylvarum

### CODE REVIEW

File: R/relObsTrends.R [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#' @examples
+relObsTrends<-function(x, timeRes, focalSp){
+  if (class(x) != "SummarizedBirds") {
+    stop("The object 'x' must be of class SummarizedBirds.")
+  }
+  if (is.null(focalSp)) {
+    stop("Please, define the focal species to search for.")
+  }
+  timeRes <- tolower(timeRes)
+  yearsAll <- as.numeric(dimnames(x$spatioTemporal)[[2]])
```
Comment (aleruete) [inline R/relObsTrends.R]: check entry

File: R/relObsTrends.R [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  allN<-summarise(group_by(spData,group),all=n())
+  spN<-summarise(group_by(spData[spData$scientificName==focalSp,],group),sp=n())
+  res<-merge(allN, spN, by="group", all.x=TRUE)
+
+  groupYMD<-strsplit(res$group,split = "-")
+  year<-as.numeric(unlist(lapply(groupYMD, function(x) x[1])))
+  month<-as.numeric(unlist(lapply(groupYMD, function(x) x[2])))
+  day<-as.numeric(unlist(lapply(groupYMD, function(x) x[3])))
```
Comment (aleruete) [inline R/relObsTrends.R]: prepare names for the xts

File: R/relObsTrends.R [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  rownames(res) <- switch(timeRes,
+                          "yearly" = paste0(year, "-01-01"),
+                          "monthly"= paste0(year, "-", sprintf("%02d", month), "-01"),
+                          "daily"  = paste0(year, "-", sprintf("%02d", month), "-", sprintf("%02d", spData$day)))
+
```
Comment (aleruete) [inline R/relObsTrends.R]: make names

File: R/relObsTrends.R [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  res<-res[,-1]
```
Comment (aleruete) [inline R/relObsTrends.R]: remove group

File: R/relObsTrends.R [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  res$relObs<-res$sp/res$all
+
+  res <- xts::as.xts(res)
```
Comment (aleruete) [inline R/relObsTrends.R]: make it an xts. Now you can quickly plot(res$sp)

Comment (aleruete) [reply R/relObsTrends.R]: Still pending the nas issue.

File: R/relObsTrends.R [hunk 6]
```diff
@@ -20,35 +20,39 @@ relObsTrends<-function(x, timeRes, focalSp){
...
 
   spData<-deconstructOverlay(x$overlaid, attr(x, "visitCol"))
-
-  spData$group<-if(timeRes=="yearly"){
-    spData$year
+  
+  spData$dates<-as.Date(switch(timeRes,
+                     "yearly" = paste0(spData$year, "-01-01"),
+                     "monthly"= paste0(spData$year, "-", sprintf("%02d", spData$month), "-01"),
+                     "daily"  = paste0(spData$year, "-", sprintf("%02d", spData$month), "-", sprintf("%02d", spData$day))))
```
Comment (antpn) [inline R/relObsTrends.R]: Moved this code to the top. We can use this function to goup the observations by!

File: R/relObsTrends.R [hunk 7]
```diff
@@ -20,35 +20,39 @@ relObsTrends<-function(x, timeRes, focalSp){
...
-    paste0(spData$year,"-",spData$month)
+    xts::xts(rep(NA, length(seq(min(spData$dates), max(spData$dates), by="month"))), 
+             seq(min(spData$dates), max(spData$dates), by="month"))
   }else if(timeRes=="daily"){
-    paste0(spData$year,"-",spData$month,"-",spData$day)
+    xts::xts(rep(NA, length(seq(min(spData$dates), max(spData$dates), by="day"))), 
+             seq(min(spData$dates), max(spData$dates), by="day"))
   }else{
     stop("Unknown time resolution")
   }
```
Comment (antpn) [inline R/relObsTrends.R]: Creates an empty xts with entries for all dates in the time span so that we have NAs for times with no data.

**Pull request merged**